bootstrap/handlers/clients.go

```diff
@@ -58,13 +59,23 @@ func (cb *ClientsBootstrap) BootstrapHandler(
 	lc := container.LoggingClientFrom(dic.Get)
 	config := container.ConfigurationFrom(dic.Get)
 	cb.registry = container.RegistryFrom(dic.Get)
-	jwtSecretProvider := secret.NewJWTSecretProvider(container.SecretProviderExtFrom(dic.Get))
 
 	if config.GetBootstrap().Clients != nil {
 		for serviceKey, serviceInfo := range *config.GetBootstrap().Clients {
 			var url string
 			var err error
 
+			sp := container.SecretProviderExtFrom(dic.Get)
+			jwtSecretProvider := secret.NewJWTSecretProvider(sp)
+			if rt, transpErr := zerotrust.HttpTransportFromClient(sp, serviceInfo, lc); transpErr != nil {
+				lc.Errorf("could not obtain an http client for use with zero trust provider: %v", transpErr)
+				return false
+			} else {
+				if sp != nil {
```

Comment:
I had it that way but the test failed again. It must be a different test had failed and I didn't notice it. I'll take a closer look. With the nil check the suite passed so I didn't dig much further. I'll look

Reply:
Thanks for the offline help. The issue was that the type assertion would fail as I'd added two new functions to the interface. Reran mockery and fixed the assertion and was able to remove the if check